src/mp-instance.js

```diff
@@ -59,7 +59,13 @@ export default function mParticleInstance(instanceName) {
     this._instanceName = instanceName;
     this._NativeSdkHelpers = new NativeSdkHelpers(this);
     this._SessionManager = new SessionManager(this);
+
+    // QUESTION: Why can't we just init the store here?
```

Comment:
line 169 shows
```
instance._Store = new Store(config, instance);
```
The store needs the config, which is an async call in a self hosted set up.  It would be possible to initialize store if store did not have items from the config.  But then we'd have an incomplete store in the beginning vs a complete store later on in the SDK which could make debugging more difficult.  Alternatively, we could put config items vs non config items in different "stores", but that woul be a larger refactor.

Reply:
The store currently has several "defaults" that are independent of the `config` and the store also contains its own `SDKConfig` object, so we technically can split this up. Perhaps we can instantiate the Store earlier and then split out the `SDKConfig` logic as its own class/object that initialized after the async config operation. Then this `SDKConfig` gets passed back to the Store after the fact, thus completing the SDK init process.

I agree that this is out of scope and I just wanted to flag this so we can think about the future refactor just so we have a plan lined up.